src/routes/game/components/dialogs/components/gameOver/GameOverDialog.vue

```diff
@@ -0,0 +1,346 @@
+      snackBarMessage: '',
+      snackColor: 'error',
+    };
+  },
+  computed: {
+    show: {
+      get() {
+        return this.modelValue;
+      },
+      set() {
+        // do nothing - parent controls whether dialog is open
+      },
+    },
+    ...mapStores(useGameStore),
+    heading() {
+      if (this.gameStore.isSpectating) {
+        return this.spectatorHeading;
+      }
+      if (this.matchIsOver) {
+        // You win the match / you lose the match
+        const messageName = this.leftPlayerWonMatch ? 'game.dialogs.gameOverDialog.youWinTheMatch' : 'game.dialogs.gameOverDialog.youLoseTheMatch';
+        return this.t(messageName);
+      }
+      // Draw / You Won / You Lose
+      const messageName = this.stalemate ? 'game.dialogs.gameOverDialog.draw' : this.playerWinsGame ? 'game.dialogs.gameOverDialog.youWin' : 'game.dialogs.gameOverDialog.youLose';
+      return this.t(messageName);
+    },
+    spectatorHeading() {
+      const prefix = 'game.dialogs.gameOverDialog';
+      if (this.gameStore.winnerPNum === null) {
+        return this.t(`${prefix}.draw`);
+      }
+      const winner = this.gameStore.players[this.gameStore.winnerPNum];
+      const res = winner.id === this.gameStore.currentMatch?.games[0].p0 ?
+        this.t(`${prefix}.p1Wins`) : this.t(`${prefix}.p2Wins`);
+      const matchTranslation = this.t(`${prefix}.match`);
+      return this.matchIsOver ? `${res} ${matchTranslation}` : res;
+    },
+    headingDataAttr() {
+      if (this.stalemate) {
+        return 'stalemate-heading';
+      }
+
+      if (this.gameStore.isSpectating) {
+        const origianlP0 = this.gameStore.currentMatch?.games[0].p0;
+        const winner = this.gameStore.players[this.gameStore.winnerPNum];
+        return winner.id === origianlP0 ? 'p1-wins-heading' : 'p2-wins-heading';
+      }
+
+      return this.playerWinsGame ? 'victory-heading' : 'loss-heading';
+    },
+    latestResult() {
+      if (this.stalemate) {
+        return 'Stalemate';
+      }
+      if (this.gameStore.isSpectating) {
+        const origianlP0 = this.gameStore.currentMatch?.games[0].p0;
+        const winner = this.gameStore.players[this.gameStore.winnerPNum];
+        return winner.id === origianlP0 ? 'Won' : 'Lost';
+      }
+
+      return this.playerWinsGame ? 'Won' : 'Lost';
+    },
+    isMobilePortrait() {
+      return this.$vuetify.display.xs;
+    },
+    currentMatch() {
+      return this.gameStore.currentMatch;
+    },
+    matchGameStats() {
+      const currentMatchGames = this.gameStore.currentMatch?.games ?? [];
+      return currentMatchGames.map((game) => {
+        if (game.status === GameStatus.FINISHED){
+          return game.winner === null ? 'D' : game.winner === this.gameStore.opponent.id ? 'L' : 'W';
+        }
+        return 'I';
+      });
+    },
+    stalemates() {
+      return this.gameStore.currentMatch?.games
+        .filter((game) => game.status === GameStatus.FINISHED && game.winner === null).length;
+    },
```

Comment:
Does this need to be a getter in the store?

Reply:
It's in a funky spot b/c it feels like something that does belong in the store (because it a value computed from the store state that seems logically like it could be used elsewhere), but it's actually only consumed in the GameOverDialog.

My inclination is to leave it here because the store is already a problematically large file so if something isn't needed as a re-used utility, then we can keep it in the consuming component (especially if it's anything smaller than GameView)

Reply:
Makes sense, I agree. I didn’t have full context so that tracks.